aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2512,6 +2572,45 @@ pub(crate) fn is_account_init_for_sponsored_transaction(
     )
 }
 
+fn get_required_deposit(
+    session: &mut SessionExt,
+    resolver: &impl AptosMoveResolver,
+    txn_gas_params: &TransactionGasParameters,
+    txn_metadata: &TransactionMetadata,
+    payload: &TransactionPayload,
+) -> Option<u64> {
+    match payload {
+        TransactionPayload::EntryFunction(entry_func) => {
+            if has_randomness_attribute(resolver, session, entry_func).unwrap_or(false) {
+                //TODO: reuse existing constants.
+                //TODO: handle overflows.
+                let internal_gas_per_gas = 1000;
```

Comment:
why not just use the scaling factor from txn_gas_params? also isn't the scaling 10^6?

Reply:
fixed